optimum/habana/transformers/generation/utils.py

```diff
@@ -957,7 +957,7 @@ def generate(
                             model_kwargs[other_inputs] = torch.nn.functional.pad(
                                 model_kwargs[other_inputs],
                                 (0, generation_config.max_new_tokens),
-                                value=generation_config.pad_token_id,
+                                value=0,
```

Comment:
@vidyasiv I am not sure setting the value to `0` is correct. What if a model uses a different padding?

Reply:
This was introduced in PR: https://github.com/huggingface/optimum-habana/pull/1276. So it used to be 0 while other paddings above it used the pad_token. It is possible this didnt need to be updated. I have noted how it fails accuracy for the models where padding is non zero above.

Reply:
I think we should look at that more accurately. My first impression is that there might be inconsistency in the code that caused these issues later.

Reply:
@zongwave please help review this part.

Reply:
Talked with @ssarkar2  Attention_mask value is either 0/1, so it will go wrong if we put non-zero pad token id.  It should be set to 0 here.

Reply:
PR where 0 was set: https://github.com/huggingface/optimum-habana/pull/488

Reply:
Yes, both attention masks and token type ids should have values in 0 and 1. So padding with `pad_token_id` was wrong. Going to merge #1371.